gui/Form1.cs

```diff
@@ -1,113 +1,194 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
 using System.ComponentModel;
 using System.Diagnostics;
 using System.Runtime.InteropServices;
-using Windows.Win32;
-using Windows.Win32.Foundation;
-using Windows.Win32.UI.Shell;
-using static System.Windows.Forms.Design.AxImporter;
+using System.Windows.Forms;
 using static System.Windows.Forms.VisualStyles.VisualStyleElement;
 using static System.Windows.Forms.VisualStyles.VisualStyleElement.Window;
+using System.Threading.Tasks;
 
 namespace gui
 {
-	public partial class Repack : Form
+	public partial class RepackForm : Form
 	{
-		List<string> files = new List<string>();
-
-		public Repack()
+		public RepackForm()
 		{
 			InitializeComponent();
+			var tip = new System.Windows.Forms.ToolTip();
+			tip.SetToolTip(lvlsList, 
+				"Choose 'Pick Folder' or drag and drop .lvl files\n"+
+				"and/or folders containing .lvls onto the list."
+				);
 		}
 
 		private void pickFolder_Click(object sender, EventArgs e)
 		{
-			PInvoke.CoCreateInstance(
-				 typeof(FileOpenDialog).GUID,
-			null,
-			Windows.Win32.System.Com.CLSCTX.CLSCTX_INPROC_SERVER,
-				 out IFileOpenDialog pickDialog).ThrowOnFailure();
+			String folderPath = null;
+			FolderBrowserDialog pickDialog = new FolderBrowserDialog();
+			pickDialog.Description = "Pick Folder";
+			if (pickDialog.ShowDialog() == DialogResult.OK)
+			{
+				folderPath = pickDialog.SelectedPath;
+			}
+			pickDialog.Dispose();
 
-			pickDialog.SetOptions(FILEOPENDIALOGOPTIONS.FOS_PATHMUSTEXIST | FILEOPENDIALOGOPTIONS.FOS_FILEMUSTEXIST | FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR | FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS);
-			pickDialog.SetTitle("Pick Folder");
-			pickDialog.SetOkButtonLabel("Pick");
+			AddPath( folderPath);
```

Comment:
Ah, I didn't realize we'd have to change this in order to use .NET Framework. The legacy folder picker is too huge a step back in usability. (For me at least.) I'd rather have the downsides of .NET 8 than it.

Reply:
So, the usability improvement would be the Drag and drop. Could use 'pick folder' dialog if you really wanted.

Reply:
Drag and Drop is a very nice addition and it provides another good workflow. For me it doesn't change the need for IFileOpenDialog to be available as well. In theory we should be able to have it both ways, use IFileOpenDialog and .NET Framework but I don't know C# well enough to know where to start with manual COM interop.

Which isn't very helpful, if I had time I would learn how and do it myself. It's probably out of scope for what you were wanting to do as well. Which I'm guessing is just to have a version people can download and have work out of the box. Towards that if it's simpler than sorting this out I don't mind at all if you'd prefer to maintain and distribute a fork without merging changes to the GUI back here.

Of course if you'd rather sort this out and get everything merged in one place I'm happy with that as well. Totally your call.